### PR TITLE
Add dependencies with v2 or bigger major versions

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

The PR adds the major version to the import paths of modules which have major versions bigger than one.

The following modules are impacted:

- `github.com/coreos/go-systemd` is updated to use `v22`
- `github.com/cespare/xxhash` is updated to use `v2`

## Why is it important?

We have a few dependencies with a major version bigger than v1. These modules include the major version in the import path. In order to include the correct dependencies, one needs to add the major version of the module to `go.mod`.

Keep in mind that every time we update a dependency to a newer major version, we need to follow-up in the import paths in our source.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made the corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

## Related issues

- Rebased on top of #15998 for the sake of testing on the CIs

### CODE REVIEW

File: journalbeat/cmd/instance/metrics.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ package instance
 import (
 	"fmt"
 
-	"github.com/coreos/go-systemd/sdjournal"
+	"github.com/coreos/go-systemd/v22/sdjournal"
```
Comment (jsoriano) [inline journalbeat/cmd/instance/metrics.go]: Is `go-systemd` changing the import path on any release? It seems rather inconvenient.

Comment (kvch) [reply journalbeat/cmd/instance/metrics.go]: It is going to be inconvenient for us every time we update a major release with go modules. At least they are adopting the modules approach, not like a few of our dependencies. :)

File: x-pack/auditbeat/module/system/host/host.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ import (
 	"strconv"
 	"time"
 
-	"github.com/cespare/xxhash"
+	xxhash "github.com/cespare/xxhash/v2"
```
Comment (urso) [inline x-pack/auditbeat/module/system/host/host.go]: is the "rename" of `v2` to `xxhash` required? Is it added by automation? If a package only imports `vX` I think one can ommit the renaming part.

Comment (kvch) [reply x-pack/auditbeat/module/system/host/host.go]: No. For some reason `goimports` kept removing the line until I added the alias. But now I am able to remove it. So I assume it was some kind of  PEBKAC. So I removed thos now.

**Comment by kvch** [comment]
No new errors has been introduced, merging.

**Pull request merged**